tests/components/opnsense/test_device_tracker.py

```diff
@@ -0,0 +1,41 @@
+"""The tests for the opnsense device tracker platform."""
+from homeassistant.setup import async_setup_component
+
+from homeassistant.components.opnsense import CONF_API_SECRET, DOMAIN, OPNSENSE_DATA
+from homeassistant.const import CONF_URL, CONF_API_KEY, CONF_VERIFY_SSL
+
+from tests.common import MockDependency, mock_coro_func
+
+FAKEFILE = None
+
+VALID_CONFIG_ROUTER_SSH = {
+    DOMAIN: {
+        CONF_URL: "https://fakehost",
+        CONF_API_KEY: "fake_key",
+        CONF_API_SECRET: "fake_secret",
+        CONF_VERIFY_SSL: False,
+    }
+}
+
+
+async def test_get_scanner(hass):
+    """Test creating an opnsense scanner."""
+    with MockDependency("pyopnsense") as mocked_opnsense:
+        mocked_opnsense.diagnostic.InterfaceClient().get_arp = mock_coro_func()
```

Comment:
We shouldn't set a mock coroutine as `get_arp` since it's not a coroutine function. Use a regular mock and have the return value of the mock return some fake data representing a seen device. Then we can assert the state of the entity at the end of the test.

Reply:
Fixed in: https://github.com/home-assistant/home-assistant/pull/26834/commits/f83386d9b87ddb3deec11414a6cb3bf955a457c5